### PR TITLE
Triangle3D performance

### PR DESCRIPTION
**Pull request by masfaraud**
* **Please check if the PR fulfills these requirements**
- [ ] The commit messages follows our guidelines (explicit, in english)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [x] Optimization: 

* Avoid calling base dict of dessia object for massively used objects
* better to_dict
* remove points attribute in edges
* Better pylint CI with time decrease

* **Other information**:

### CODE REVIEW

File: volmdlr/edges.py [hunk 1]
```diff
@@ -1216,22 +1219,22 @@ def create_tangent_circle(self, point, other_line):
...
 
         if math.isclose(self.point_distance(point), 0, abs_tol=1e-10):
-            I = volmdlr.Vector2D(point[0], point[1])
-            A = volmdlr.Vector2D(self.points[0][0], self.points[0][1])
-            B = volmdlr.Vector2D(self.points[1][0], self.points[1][1])
-            C = volmdlr.Vector2D(other_line.points[0][0],
-                                 other_line.points[0][1])
-            D = volmdlr.Vector2D(other_line.points[1][0],
-                                 other_line.points[1][1])
+            I = volmdlr.Vector2D(point[0], point.y)
```
Comment (masfaraud) [inline volmdlr/edges.py:1222]: .x

**Review by WirajanDASILVA — APPROVED** [review]

File: volmdlr/edges.py [hunk 2]
```diff
@@ -3920,12 +3923,12 @@ def plot_data(self, x_3D, y_3D, marker=None, color='black', stroke_width=1,
...
-        x1, y1, z1 = round(1000 * self.start, ndigits)
-        x2, y2, z2 = round(1000 * self.end, ndigits)
-        return '{} = Part.LineSegment(fc.Vector({},{},{}),fc.Vector({},{},{}))\n'.format(
-            name, x1, y1, z1, x2, y2, z2)
+    # def FreeCADExport(self, name, ndigits=6):
+    #     name = 'primitive' + str(name)
+    #     x1, y1, z1 = round(1000 * self.start, ndigits)
+    #     x2, y2, z2 = round(1000 * self.end, ndigits)
+    #     return '{} = Part.LineSegment(fc.Vector({},{},{}),fc.Vector({},{},{}))\n'.format(
+    #         name, x1, y1, z1, x2, y2, z2)
```
Comment (WirajanDASILVA) [inline volmdlr/edges.py:3931]: Is this still important, if not could you please delete it?

File: volmdlr/edges.py [hunk 3]
```diff
@@ -4363,26 +4366,26 @@ def f(param):
...
+    #     # !!! : A QUOI SERT LE DERNIER ARG DE BSplineCurve (False)?
+    #     # LA MULTIPLICITE EN 3e ARG ET LES KNOTS EN 2e ARG ?
+    #     return '{} = Part.BSplineCurve({},{},{},{},{},{},{})\n'.format(name,
+    #                                                                    points,
+    #                                                                    self.knot_multiplicities,
+    #                                                                    self.knots,
+    #                                                                    self.periodic,
+    #                                                                    self.degree,
+    #                                                                    self.weights,
+    #                                                                    False)
```
Comment (WirajanDASILVA) [inline volmdlr/edges.py:4388]: delete it?

File: volmdlr/edges.py [hunk 4]
```diff
@@ -5128,12 +5129,12 @@ def plot2d(self, center: volmdlr.Point3D = volmdlr.O3D,
...
-        xi, yi, zi = round(1000 * self.interior, ndigits)
-        xe, ye, ze = round(1000 * self.end, ndigits)
-        return '{} = Part.Arc(fc.Vector({},{},{}),fc.Vector({},{},{}),fc.Vector({},{},{}))\n' \
-            .format(name, xs, ys, zs, xi, yi, zi, xe, ye, ze)
+    # def FreeCADExport(self, name, ndigits=6):
+    #     xs, ys, zs = round(1000 * self.start, ndigits)
+    #     xi, yi, zi = round(1000 * self.interior, ndigits)
+    #     xe, ye, ze = round(1000 * self.end, ndigits)
+    #     return '{} = Part.Arc(fc.Vector({},{},{}),fc.Vector({},{},{}),fc.Vector({},{},{}))\n' \
+    #         .format(name, xs, ys, zs, xi, yi, zi, xe, ye, ze)
```
Comment (WirajanDASILVA) [inline volmdlr/edges.py:5137]: delete it?

File: volmdlr/edges.py [hunk 5]
```diff
@@ -5981,23 +5980,23 @@ def plot2d(self, x3d: volmdlr.Vector3D = volmdlr.X3D, y3d: volmdlr.Vector3D = vo
...
-        xi, yi, zi = round(1000 * self.interior, ndigits).vector
-        xe, ye, ze = round(1000 * self.end, ndigits).vector
-        return '{} = Part.Arc(fc.Vector({},{},{}),fc.Vector({},{},{}),fc.Vector({},{},{}))\n'.format(
-            name, xs, ys, zs, xi, yi, zi, xe, ye, ze)
+    # def FreeCADExport(self, name, ndigits=6):
+    #     xs, ys, zs = round(1000 * self.start, ndigits).vector
+    #     xi, yi, zi = round(1000 * self.interior, ndigits).vector
+    #     xe, ye, ze = round(1000 * self.end, ndigits).vector
+    #     return '{} = Part.Arc(fc.Vector({},{},{}),fc.Vector({},{},{}),fc.Vector({},{},{}))\n'.format(
+    #         name, xs, ys, zs, xi, yi, zi, xe, ye, ze)
```
Comment (WirajanDASILVA) [inline volmdlr/edges.py:5999]: delete it?

**Review by WirajanDASILVA — CHANGES_REQUESTED** [review]

File: volmdlr/edges.py (hunk 5 above)
Comment (masfaraud) [reply volmdlr/edges.py]: done

**Review by WirajanDASILVA — APPROVED** [review]

**Pull request merged**